src/ray/object_manager/object_store_pool.h

```diff
@@ -0,0 +1,62 @@
+#ifndef RAY_OBJECT_STORE_POOL_H
+#define RAY_OBJECT_STORE_POOL_H
+
+#include <list>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+#include <boost/asio.hpp>
+#include <boost/asio/error.hpp>
+#include <boost/bind.hpp>
+
+#include "plasma/client.h"
+#include "plasma/events.h"
+#include "plasma/plasma.h"
+
+#include "object_directory.h"
+#include "ray/id.h"
+#include "ray/status.h"
+
+namespace ray {
+
+class ObjectStorePool {
+ public:
+  /// Provides connections to the object store. Enables concurrent communication with
+  /// the object store.
+  ///
+  /// \param store_socket_name The object store socket name.
+  ObjectStorePool(std::string &store_socket_name);
+
+  /// This object cannot be copied due to pool_mutex.
+  ObjectStorePool &operator=(const ObjectStorePool &o) {
```

Comment:
better use the RAY_DISALLOW_COPY_AND_ASSIGN macro from https://github.com/ray-project/ray/blob/master/src/ray/util/macros.h